WalletWasabi/Blockchain/TransactionBuilding/ChangelessTransactionCoinSelector.cs

```diff
@@ -0,0 +1,55 @@
+using NBitcoin;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using WalletWasabi.Blockchain.TransactionBuilding.BnB;
+using WalletWasabi.Blockchain.TransactionOutputs;
+
+namespace WalletWasabi.Blockchain.TransactionBuilding;
+
+public static class ChangelessTransactionCoinSelector
+{
+	public static bool TryGetCoins(List<SmartCoin> availableCoins, FeeRate feeRate, long target, [NotNullWhen(true)] out List<SmartCoin>? selectedCoins, CancellationToken cancellationToken = default)
+	{
+		selectedCoins = null;
+		// Keys are effective values of smart coins in satoshis.
+		IOrderedEnumerable<SmartCoin> sortedCoins = availableCoins.OrderByDescending(x => x.EffectiveValue(feeRate).Satoshi);
+
+		Dictionary<SmartCoin, long> inputs = new(sortedCoins.ToDictionary(x => x, x => x.EffectiveValue(feeRate).Satoshi));
+
+		// Pass smart coins' effective values in ascending order.
```

Comment:
`in ascending order`? Isn't the `sortedCoins` in desc?

Reply:
Yes, this is incorrect.

Reply:
Thanks